bciers/apps/registration/tests/components/form/widgets/TextWidget.test.tsx

```diff
@@ -0,0 +1,159 @@
+import { userEvent } from "@testing-library/user-event";
+import { render, screen } from "@testing-library/react";
+import { RJSFSchema } from "@rjsf/utils";
+import FormBase from "@/app/components/form/FormBase";
+import { checkTextWidgetValidationStyles } from "@/tests/helpers/form";
+
+const stringFieldLabel = "String test field";
+const numberFieldLabel = "Number test field";
+const stringLabelRequired = `${stringFieldLabel}*`;
+const numberLabelRequired = `${numberFieldLabel}*`;
+
+export const stringFieldSchema = {
+  type: "object",
+  required: ["stringTestField"],
+  properties: {
+    stringTestField: { type: "string", title: stringFieldLabel },
+  },
+} as RJSFSchema;
+
+export const numberFieldSchema = {
+  type: "object",
+  required: ["numberTestField"],
+  properties: {
+    numberTestField: { type: "number", title: numberFieldLabel },
+  },
+} as RJSFSchema;
+
+const numberFieldUiSchemaWithMax = {
+  numberTestField: {
+    "ui:options": {
+      max: 10,
+    },
+  },
+};
+
+describe("RJSF TextWidget", () => {
+  it("should render a text field", () => {
+    render(<FormBase schema={stringFieldSchema} />);
+    expect(screen.getByLabelText(stringLabelRequired)).toBeVisible();
+  });
+
+  it("should render a number field", () => {
+    render(<FormBase schema={numberFieldSchema} />);
+    expect(screen.getByLabelText(numberLabelRequired)).toBeVisible();
+  });
+
+  it("should be empty by default", () => {
+    render(<FormBase schema={stringFieldSchema} />);
+    const input = screen.getByLabelText(stringLabelRequired);
+    expect(input).toHaveValue("");
+  });
+
+  it("should render the string value when formData is provided", () => {
+    render(
+      <FormBase
+        schema={stringFieldSchema}
+        formData={{ stringTestField: "test" }}
+      />,
+    );
+
+    const input = screen.getByLabelText(stringLabelRequired);
+    expect(input).toHaveValue("test");
+  });
+
+  it("should render the number value when formData is provided", () => {
+    render(
+      <FormBase
+        schema={numberFieldSchema}
+        formData={{ numberTestField: 123 }}
+      />,
+    );
+
+    const input = screen.getByLabelText(numberLabelRequired);
+    expect(input).toHaveValue(123);
+  });
+
+  it("should allow entering text", async () => {
+    render(<FormBase schema={stringFieldSchema} />);
+    const input = screen.getByLabelText(stringLabelRequired);
+    await userEvent.type(input, "test");
+    expect(input).toHaveValue("test");
+  });
+
+  it("should only allow entering numbers in a number field", async () => {
```

Comment:
I think the other use case we might need in the future is decimal values (e.g., 23.5)
and in the case of reporting in particular, I'd expect there will be users typing in large values like "1,320,000" so maybe a test with commas in the input value?

Reply:
Sounds good add a `should not allow entering commas in a number field` test.